fabric-crash-report-info-v1/src/main/java/net/fabricmc/fabric/mixin/crash/report/info/MixinCrashReport.java

```diff
@@ -39,22 +39,36 @@ public abstract class MixinCrashReport {
 	@Inject(at = @At("RETURN"), method = "<init>")
 	private void fillSystemDetails(CallbackInfo info) {
 		addSection("Fabric Mods", () -> {
-			Map<String, String> mods = new TreeMap<>();
+			ArrayList<ModContainer> topLevelMods = new ArrayList<>();
 
 			for (ModContainer container : FabricLoader.getInstance().getAllMods()) {
-				mods.put(container.getMetadata().getId(), container.getMetadata().getName() + " " + container.getMetadata().getVersion().getFriendlyString());
+				if (container.getContainingMod().isEmpty()) {
+					topLevelMods.add(container);
+				}
 			}
 
 			StringBuilder modString = new StringBuilder();
 
-			for (String id : mods.keySet()) {
-				modString.append("\n\t\t");
-				modString.append(id);
-				modString.append(": ");
-				modString.append(mods.get(id));
-			}
+			appendMods(modString, 2, topLevelMods);
 
 			return modString.toString();
 		});
 	}
+
+	private static void appendMods(StringBuilder modString, int depth, ArrayList<ModContainer> mods) {
+		mods.sort(Comparator.comparing(mod -> mod.getMetadata().getId()));
+
+		for (ModContainer mod: mods) {
```

Comment:
```suggestion
		for (ModContainer mod : mods) {
```
How come this passes the checkstyle check? 🤔

Reply:
I don't know. I fear checkstyle. I'm happy as long as it doesn't cause the build to fail. Changed it.